src/main/java/chess/web/service/ChessService.java

```diff
@@ -1,106 +1,169 @@
+                .orElseThrow(() -> new IllegalArgumentException("없는 체스방 id 입니다."));
+        roomDao.updateTurnById(roomId, Turn.init().getTeam().value());
+    }
 
-        pieceDao.deleteByBoardId(boardId);
+    private void initPiece(Long roomId) {
+        pieceDao.deleteAllByRoomId(roomId);
 
-        boardDao.updateTurnById(boardId, turn.getTeam().value());
+        Pieces pieces = Pieces.createInit();
         for (Piece piece : pieces.getPieces()) {
-            insertOrUpdatePiece(boardId, piece);
+            insertOrUpdatePiece(roomId, piece);
         }
-
-        Pieces savedPieces = Pieces.from(pieceDao.findAllByBoardId(boardId));
-        return Board.create(savedPieces, turn);
     }
 
-    private void insertOrUpdatePiece(Long boardId, Piece piece) {
+    private void insertOrUpdatePiece(Long roomId, Piece piece) {
         String position = piece.getPosition().name();
         String type = piece.getType();
         String team = piece.getTeam().value();
 
-        if (pieceDao.findByPositionAndBoardId(position, boardId).isPresent()) {
-            pieceDao.updatePieceByPositionAndBoardId(type, team, position, boardId);
+        if (pieceDao.findByPositionAndRoomId(position, roomId).isPresent()) {
+            pieceDao.updatePieceByPositionAndRoomId(type, team, position, roomId);
             return;
         }
-        pieceDao.save(piece, boardId);
+        pieceDao.save(piece, roomId);
     }
 
-    public ScoreDto getStatus(Long boardId) {
-        List<Piece> found = pieceDao.findAllByBoardId(boardId);
+    @Transactional(readOnly = true)
+    public ScoreDto getStatus(Long roomId) {
+        List<Piece> found = pieceDao.findAllByRoomId(roomId);
         Pieces pieces = Pieces.from(found);
 
         double blackScore = pieces.getTotalScore(Team.BLACK);
         double whiteScore = pieces.getTotalScore(Team.WHITE);
         return new ScoreDto(blackScore, whiteScore);
     }
+
+    @Transactional
+    public Long createRoom(String title, String password) {
+        Long id = roomDao.save(Turn.init().getTeam().value(), title, password);
+        pieceDao.save(Pieces.createInit().getPieces(), id);
+        return id;
+    }
+
+    @Transactional(readOnly = true)
+    public List<Room> getRooms() {
+        return roomDao.findAll();
+    }
+
+    @Transactional
+    public boolean removeRoom(Long roomId, String password) {
+        if (!canRemoveRoom(roomId, password)) {
+            return false;
+        }
+        pieceDao.deleteAllByRoomId(roomId);
+        roomDao.delete(roomId, password);
+        return true;
+    }
+
+    private boolean canRemoveRoom(Long roomId, String password) {
+        Optional<Room> optionalRoom = roomDao.findById(roomId);
+        if (optionalRoom.isEmpty()) {
+            return false;
+        }
+        Room room = optionalRoom.get();
+        if (isRunningChess(roomId, room) || !matchPassword(room, password)) {
+            return false;
+        }
+        return true;
```

Comment:
```suggestion
        return !isRunningChess(roomId, room) && matchPassword(room, password);
```

아래와 같이 바꿀 수 있겠네요!

해당 라인을 메서드로 추출하면 더 가독성이 좋아지겠죠?

Reply:
네 수정했습니다!